catalyst-gateway/bin/src/cardano/mod.rs

```diff
@@ -428,11 +414,13 @@ struct SyncTask {
     /// Current Sync Status.
     sync_status: Vec<SyncStatus>,
 
-    /// Event sender during the process of sync tasks.
-    event_channel: (
-        broadcast::Sender<event::ChainIndexerEvent>,
-        broadcast::Receiver<event::ChainIndexerEvent>,
-    ),
+    /// A channel that contains an end slot number for every immutable sync task.
+    ///
+    /// For example, if there are 3 tasks with `0..100`, `100..200` and `200..300` block
+    /// ranges (where the end bounds aren't inclusive), then the channel will contain the
+    /// following list: `[99, 199, 299]`. A slot value is removed when that task is done,
+    /// so when the second task is finished the list will be updated to `[99, 299]`.
+    pending_blocks: watch::Sender<BTreeSet<Slot>>,
```

Comment:
This is not related to the purpose of the PR and should not be merged in this branch, because it is a new functionality
```suggestion
```